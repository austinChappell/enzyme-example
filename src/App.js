// External Dependencies
import React, { Component } from 'react';
import styled from 'styled-components';

// Internal Dependencies
import Button from './components/Button';
import Input from './components/Input';

// Local Variables
const ContentWrapper = styled.div({
  paddingTop: 100,
  textAlign: 'center',
})
const InputWrapper = styled.div({
  marginBottom: 12,
})

// Component Definition
class App extends Component {
  state = {
    email: '',
    password: '',
  }

  handleClickButton = () => {
    console.log('button clicked');
  }

  handleChangeInput = (e) => {
    console.log(e.target.name)
    this.setState({
      [e.target.name]: e.target.value,
    })
  }

  render() {
    return (
      <ContentWrapper>
        <InputWrapper>
          <Input
            ariaLabel="email"
            name="email"
            onChange={this.handleChangeInput}
            placeholder="Email"
            value={this.state.email}
          />
        </InputWrapper>

        <InputWrapper>
          <Input
            ariaLabel="password"
            name="password"
            onChange={this.handleChangeInput}
            placeholder="Password"
            type="password"
            value={this.state.password}
          />
        </InputWrapper>

        <Button onClick={this.handleClickButton}>
          Sign Up
        </Button>
      </ContentWrapper>
    );
  }
}

export default App;
