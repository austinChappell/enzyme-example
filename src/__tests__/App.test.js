// External Dependencies
import React from 'react';
import { render, shallow } from 'enzyme';

// Internal Dependencies
import App from '../App';

describe('<App />', () => {
  it('will render', () => {
    const wrapper = shallow(
      <App />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('contains "Enzyme Testing Demo', () => {
    const wrapper = render(
      <App />,
    );
    expect(wrapper.text()).toContain('Enzyme Testing Demo');
  });
});
