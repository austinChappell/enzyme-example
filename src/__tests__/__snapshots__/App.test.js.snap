// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> will render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Drawer />,
        <ForwardRef>
          <h2>
            Enzyme Testing Demo
          </h2>
          <ForwardRef>
            <Input
              ariaLabel="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
          </ForwardRef>
          <ForwardRef>
            <Input
              ariaLabel="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </ForwardRef>
          <Button
            disabled={false}
            onClick={[Function]}
          >
            Sign Up
          </Button>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h2>
              Enzyme Testing Demo
            </h2>,
            <ForwardRef>
              <Input
                ariaLabel="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />
            </ForwardRef>,
            <ForwardRef>
              <Input
                ariaLabel="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </ForwardRef>,
            <Button
              disabled={false}
              onClick={[Function]}
            >
              Sign Up
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enzyme Testing Demo",
            },
            "ref": null,
            "rendered": "Enzyme Testing Demo",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                ariaLabel="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "ariaLabel": "email",
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "margin-bottom: 12px;",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                ariaLabel="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "ariaLabel": "password",
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": true,
                "rules": Array [
                  "margin-bottom: 12px;",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Sign Up",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "padding-top: 100px; text-align: center;",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Drawer />,
          <ForwardRef>
            <h2>
              Enzyme Testing Demo
            </h2>
            <ForwardRef>
              <Input
                ariaLabel="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
              />
            </ForwardRef>
            <ForwardRef>
              <Input
                ariaLabel="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </ForwardRef>
            <Button
              disabled={false}
              onClick={[Function]}
            >
              Sign Up
            </Button>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h2>
                Enzyme Testing Demo
              </h2>,
              <ForwardRef>
                <Input
                  ariaLabel="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />
              </ForwardRef>,
              <ForwardRef>
                <Input
                  ariaLabel="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </ForwardRef>,
              <Button
                disabled={false}
                onClick={[Function]}
              >
                Sign Up
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enzyme Testing Demo",
              },
              "ref": null,
              "rendered": "Enzyme Testing Demo",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Input
                  ariaLabel="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  type="text"
                  value=""
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "ariaLabel": "email",
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "Email",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "rules": Array [
                    "margin-bottom: 12px;",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Input
                  ariaLabel="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "ariaLabel": "password",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": true,
                  "rules": Array [
                    "margin-bottom: 12px;",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Sign Up",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "padding-top: 100px; text-align: center;",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
